clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -1246,7 +1251,8 @@ static void WarnForDeprecatedBackendOpts(const Driver &D,
   // Make sure to only warn for once for gen targets as the translate
   // options tree is called twice but only the second time has the
   // device set.
-  if (Triple.isSPIR() && Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen &&
+  if ((Triple.isSPIR() || Triple.isSPIRV()) &&
+      Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen &&
```

Comment:
This addition is out of place.  We don't have `spirv64_gen` as a arch_subarch variant at this time.

Reply:
Removed isSPIRV() check. Thanks for catching this.